src/pdf.rs

```diff
@@ -0,0 +1,673 @@
+                current_layer.use_text(current_page_text, 13.0, Mm(185.0), Mm(14.0), &font_medium);
+
+                // Call the refactored function to add a new page
+                let (new_layer, _new_y_pos, new_page) =
+                    add_new_page(&doc, current_page, &font_bold, &column_widths, line_height);
+
+                // Update current_layer, current_y_pos, and current_page with the returned values
+                current_layer = new_layer;
+                current_y_pos = Mm(273.0) - line_height;
+                current_page = new_page;
+            }
+
+            let work_time_str = format_minutes_as_time(*work_time);
+            let ride_time_str = format_minutes_as_time(*ride_time);
+
+            // Calculate the total time (work + ride)
+            let total_time = work_time + ride_time;
+            let total_time_str = format_minutes_as_time(total_time);
+
+            let points = vec![
+                (Point::new(Mm(21.0), current_y_pos - Mm(3.0)), false),
+                (
+                    Point::new(Mm(21.0), current_y_pos + line_height - Mm(3.0)),
+                    false,
+                ),
+                (
+                    Point::new(Mm(188.0), current_y_pos + line_height - Mm(3.0)),
+                    false,
+                ),
+                (Point::new(Mm(188.0), current_y_pos - Mm(3.0)), false),
+            ];
+            let rectangle = Line {
+                points,
+                is_closed: true,
+                has_fill: true,
+                has_stroke: false,
+                is_clipping_path: false,
+            };
+            let header_colour = Color::Rgb(Rgb::new(0.952, 0.952, 0.952, None));
+            current_layer.set_fill_color(header_colour);
+            current_layer.add_shape(rectangle);
+
+            let text_colour = Color::Rgb(Rgb::new(0.0, 0.0, 0.0, None));
+            current_layer.set_fill_color(text_colour);
+
+            // Output task_id rows
+            current_layer.use_text("", font_size, column_widths[0], current_y_pos, &font_medium); // Empty first column
+            current_layer.use_text(
+                &work_time_str,
+                font_size,
+                column_widths[1],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &ride_time_str,
+                font_size,
+                column_widths[2],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &total_time_str,
+                font_size,
+                column_widths[3],
+                current_y_pos,
+                &font_medium,
+            );
+            current_layer.use_text(
+                &*task_id,
+                font_size,
+                column_widths[4],
+                current_y_pos,
+                &font_medium,
+            );
+
+            current_y_pos -= line_height;
```

Comment:
here has to be something off. Because the rows are not filling completely. Every new row there is a tiny small line in between if you look carefully. Like you REALLY have to look closely.
![image](https://github.com/user-attachments/assets/f18ac46a-9adc-40eb-8163-5a6508db56dd)

Reply:
isn feature